### PR TITLE
DB-9682 Restrict columns for CSV external tables to be exact

### PR DESCRIPTION
**Pull request by martinrupp**

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1492/

**Review by jyuanca — APPROVED** [review]

**Review by yxia92 — APPROVED** [review]
+1

**Review by ascend1 — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

File: hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java [hunk 1]
```diff
@@ -1010,6 +1010,23 @@ public void testReadTextMismatchSchema() throws Exception {
...
+            Assert.fail("Exception not thrown");
+        } catch (SQLException e) {
+            Assert.assertEquals("Wrong Exception (" + e.getMessage() + ")", EXTERNAL_TABLES_READ_FAILURE, e.getSQLState());
+            Assert.assertEquals( "wrong exception message",
+                    "External Table read failed with exception '2 attribute(s) defined but 3 present " +
+                            "in the external file : '" + file + "'. Suggested Schema is 'CREATE EXTERNAL TABLE T " +
+                            "(_c0 CHAR/VARCHAR(x), _c1 CHAR/VARCHAR(x), _c2 CHAR/VARCHAR(x));'.'",
+                    e.getMessage() );
+        }
+        methodWatcher.execute("drop table external_t3");
```
Comment (hatyo) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java:1029]: this should be executed regardless if the test fails or not, think about restructuring the test accordingly (e.g. use `finally`).

Comment (martinrupp) [reply hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: Hi you're correct for other tests, however here we create many different tables, adding them to the preamble would mean creating 50 tables there, which is - in context of external tables - not really great for overview.

Comment (hatyo) [reply hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: ok :)

**Review by hatyo — APPROVED** [review]

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by dgomezferro — APPROVED** [review]

**Pull request merged**